### DISCUSSION TITLE
[4.0.0-beta.1] UI for editing focal point in preview modal

### DISCUSSION THREAD

**Discussion by wsydney76** [craftcms/cms#10763] [Ideas]
From #10743 

> I assume the UI for this is still in development? Clicking once on the thumbnail opening the modal, double clicking outside opening a slideout is not very intuitive, and there is no visual feedback when you changed the focal point in the modal.

Noticed a user (well, me) was completly lost trying to figure out how to open the slideout editor from an assets field.

> The UI might be tweaked down the road - can you please start a new discussion about that, though?

Here it is.

I like how it's done in another CMS

* one action dropdown for all actions
* a default action (open) triggered by a click/dblclick/return key
* a drag handle when hovering over the image

![image](https://user-images.githubusercontent.com/60010597/158634215-890ea80a-d6f7-44ed-b492-adc03cd82dc7.png)

Actions in Craft could be (if  a user has the appropriate permission)

* Edit (in slideout)
* Open image editor (with #8489 implemented?)
* Set focal point (in preview modal)
* Remove relationship

It's not that fancy than things like buttons appearing/disappearing when hovering over an image, but it's crystal clear, and i think it's easier to make it keyboard accessible. 

Prevents users like me from accidentally removing a relationship, because it takes two clicks to do that.

And it opens up the possibility to add custom actions via an event.

The preview modal itself:

* Would like to see a tool bar with always visible buttons like enable/disable focal point, close modal, undo to previous state
* Give a visual feedback when a focal point was saved
* In live preview: I think it would make sense to constrain the modal to the left editing pane, so you can see the impact immediatly in the preview without closing the modal.

**Comment by MoritzLost** [comment]
+1000, the new UX is really unintuitive. First of, there's no indication that clicking / double clicking opens anything at all, which most of my clients initially have trouble figuring out – this was already an issue in Craft 3. But now its even worse, because clicking on the image opens the focal point editor and the _only_ way to open the asset modal is to double click the title of the image:

![Screenshot of the asset selector in the control panel](https://user-images.githubusercontent.com/10146880/167429243-beafe284-f853-4b33-b91e-82ea1aa44e76.png)

This is impossible to figure out on your own.

I like the drop-down menu idea, that gives it room to grow for more options / interactions. Another option to make the most important interactions clearly visible:

![Screenshot of Craft's selected asset view with added links to edit the asset and/or the focal points](https://user-images.githubusercontent.com/10146880/167429482-1a38bdbf-b1ed-4d68-914f-d72be40b79b8.png)

Super quick mockup, obviously this needs some polishing. But something like this would immensely improve the UX.

**Comment by gcamacho079** [comment]
Hey @MoritzLost, apologies for the late response! We agree that the UI for opening a slideout editor needs to be more apparent and be keyboard accessible.

We’re working on a solution to make this functionality accessible and have a PR in review that tackles this particular issue (see PR #12660); we’re also discussing the potential for a more comprehensive Action dropdown, and the other UX points raised regarding the preview modal itself.

**Reply by MoritzLost** [reply]
@gcamacho079 Thanks, that sounds great! Yeah, an edit button would be great. Hoping to see this merged soon!
